server/src/main/java/org/opensearch/common/remote/AbstractClusterMetadataWriteableBlobEntity.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.remote;
+
+import org.opensearch.core.compress.Compressor;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.gateway.remote.ClusterMetadataManifest.UploadedMetadata;
+
+/**
+ * An extension of {@link RemoteWriteableEntity} class which caters to the use case of writing to and reading from a blob storage
+ *
+ * @param <T> The class type which can be uploaded to or downloaded from a blob storage.
+ */
+public abstract class AbstractClusterMetadataWriteableBlobEntity<T> extends RemoteWriteableBlobEntity<T> {
+
+    protected final NamedXContentRegistry namedXContentRegistry;
+
+    public AbstractClusterMetadataWriteableBlobEntity(
+        final String clusterUUID,
+        final Compressor compressor,
+        final NamedXContentRegistry namedXContentRegistry
+    ) {
+        super(clusterUUID, compressor);
+        this.namedXContentRegistry = namedXContentRegistry;
+    }
+
+    public AbstractClusterMetadataWriteableBlobEntity(final String clusterUUID, final Compressor compressor) {
+        super(clusterUUID, compressor);
+        this.namedXContentRegistry = null;
+    }
+
+    public abstract UploadedMetadata getUploadedMetadata();
```

Comment:
UploadedMetadata is very generic. It just provides the component name and the uploaded filename. If we need to know the uploaded remote path, then this can be used.
Should we move it out of ClusterMetadataManifest to commons and then include in the RemoteWritableBlobEntity class?

Reply:
See https://github.com/opensearch-project/OpenSearch/pull/15180 for reference, this was being set to `null`

Reply:
Got it. But I think it would good to return this attribute as well to know the blob name which got uploaded.

Reply:
Yes I need to think more on other parts of the code needing to upload a manifest. Will take that up need basis as a follow-up. Thanks